### PR TITLE
Don't use `UrlFetchTransport` in App Engine Flex environment 

### PR DESCRIPTION
**Pull request by vam-google**
https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1492

### CODE REVIEW

**Comment by garrettjonesgoogle** [comment]
@shinfan can you do the first pass?

File: testing/google-cloud-managed-test/src/main/java/com/google/cloud/managed/test/GaeGcjTestServlet.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 import javax.servlet.http.HttpServletResponse;
 
 @WebServlet(name = "flex", value = "/test")
```
Comment (shinfan) [inline testing/google-cloud-managed-test/src/main/java/com/google/cloud/managed/test/GaeGcjTestServlet.java]: rename flex to gae?

File: testing/google-cloud-managed-test/src/main/java/com/google/cloud/managed/test/GaeInfoServlet.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Map;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "hex", value = "/info")
```
Comment (shinfan) [inline testing/google-cloud-managed-test/src/main/java/com/google/cloud/managed/test/GaeInfoServlet.java]: hex?

File: testing/google-cloud-appengine-java8/pom.xml [hunk 3]
```diff
@@ -41,7 +41,7 @@
 
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <version>3.3</version>
+        <version>3.5.1</version>
```
Comment (garrettjonesgoogle) [inline testing/google-cloud-appengine-java8/pom.xml]: What impact does this have?

File: testing/google-cloud-managed-test/src/main/java/com/google/cloud/managed/test/GaeInfoServlet.java [hunk 4]
```diff
@@ -33,10 +35,20 @@
...
       throws ServletException, IOException {
     resp.setContentType("text");
-
     PrintWriter out = resp.getWriter();
-    out.append("\nENVIRONMENT VARIABLES:\n\n");
 
+    ClassLoader cl = getClass().getClassLoader();
+    out.append("CLASS LOADER:\n\n");
+    out.append("Name: ").append(cl != null ? cl.getClass().getName() : null).append('\n');
+    if(cl instanceof URLClassLoader) {
```
Comment (garrettjonesgoogle) [inline testing/google-cloud-managed-test/src/main/java/com/google/cloud/managed/test/GaeInfoServlet.java]: use a space after keywords (`if`, `for`)

**Comment by vam-google** [comment]
PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**